pkg/dockerregistry/server/auth.go

```diff
@@ -147,6 +151,36 @@ var (
 	ErrUnsupportedResource = errors.New("unsupported resource")
 )
 
+// TokenRealm returns the template URL to use as the token realm redirect.
+// An empty scheme/host in the returned URL means to match the scheme/host on incoming requests.
+func TokenRealm(options map[string]interface{}) (*url.URL, error) {
+	if options[TokenRealmKey] == nil {
+		// If not specified, default to "/openshift/token", auto-detecting the scheme and host
+		return &url.URL{Path: defaultTokenPath}, nil
+	}
+
+	tokenRealmString, ok := options[TokenRealmKey].(string)
+	if !ok {
+		return nil, fmt.Errorf("%s config option must be a string, got %T", TokenRealmKey, options[TokenRealmKey])
+	}
+
+	tokenRealm, err := url.Parse(tokenRealmString)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing URL in %s config option: %v", TokenRealmKey, err)
+	}
+	if len(tokenRealm.RawQuery) > 0 || len(tokenRealm.Fragment) > 0 {
+		return nil, fmt.Errorf("%s config option may not contain query parameters or a fragment", TokenRealmKey)
+	}
+	if len(tokenRealm.Path) > 0 {
+		return nil, fmt.Errorf("%s config option may not contain a path (%q was specified)", TokenRealmKey, tokenRealm.Path)
+	}
+
+	// pin to "/openshift/token"
+	tokenRealm.Path = defaultTokenPath
```

Comment:
Started out being able to be overridden, but I couldn't come up with a reasonable way to know what paths to disallow.